internal/walk/walklinks.go

```diff
@@ -31,6 +46,22 @@ func walkDirWithLinks(ctx context.Context, root string, fn fs.WalkDirFunc) error
 			return ctx.Err()
 		}
 
+		for _, str := range exclude {
+			re, err := makeRegexp(str)
+			if err != nil {
+				return err
```

Comment:
I think this is a case where, if there's an error, we should really wrap it before returning.  It seems like the error will be rather hard to understand otherwise because the `str` here we pass into Compile isn't equal to what the user passed on the command-line (e.g. if user passed "foo*bar", the error message may instead mention "foo[^/]+bar" and the user isn't going to know why).

If the error is non-nil we should probably wrap it with something like `fmt.Errorf("--exclude '%s': %w", str, err)` so the user can trace it back to an --exclude argument.

https://github.com/release-engineering/exodus-rsync/wiki/Coding-guidelines#error-handling

I suppose the wrapping could also happen inside of makeRegexp but here seems more convenient since you're already testing `err`.

Reply:
Good call.